.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Image Build and Publish
+
+on:
+  push:
+    branches: [ "main" ]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+    - name: Extract metadata (tags, labels) for Image
+      id: meta
+      uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5
+      with:
+        context: .
+        platforms: linux/amd64,linux/arm64,linux/arm/v6,linux/arm/v7,linux/arm64/v8
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
```

Comment:
What exactly would be the tag in this case? It seems that what we need instead is to add the git tag.

Reply:
See https://github.com/docker/metadata-action?tab=readme-ov-file#usage for more details. It depends on the github events that is triggering the workflow. The intend here is to react on the push tag and pust to the main branch events for releasing new versions and the `latest` one

Reply:
Thank you. I haven't used `build-push-action` action before but it looks useful.
According to the docs we are going to have:
```
push		refs/heads/main		main
push tag	refs/tags/v1.2.3	v1.2.3, latest
```
Let's keep just the `push tag` event triggering the release to avoid new images on unrelated commits like docs for example.

Reply:
done, adjusted it to trigger the release on push tags events only.